integration/map.go

```diff
@@ -17,28 +17,63 @@ package integration
 import (
 	"bytes"
 	"context"
-	"crypto"
 	"encoding/hex"
 	"fmt"
 	"math/rand"
 
 	"github.com/golang/glog"
 	"github.com/google/trillian"
 	tcrypto "github.com/google/trillian/crypto"
+	"github.com/google/trillian/crypto/keys"
 	"github.com/google/trillian/merkle"
-	"github.com/google/trillian/merkle/maphasher"
+	"github.com/google/trillian/merkle/hashers"
 	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/testonly/integration"
 )
 
-const treeID = int64(0)
+// createHashKV returns a []*trillian.MapLeaf formed by the mapping of index, value ...
+// createHashKV panics if len(iv) is odd. Duplicate i/v pairs get over written.
+func createMapLeaves(iv ...[]byte) []*trillian.MapLeaf {
```

Comment:
`trillian.MapLeaf` is exactly such a struct.  I'll adjust the `makeBatchIndexValuePairs` just emit a slice of `MapLeaf`s